xlators/storage/posix/src/posix-helpers.c

```diff
@@ -2276,7 +2276,12 @@ posix_spawn_health_check_thread(xlator_t *xl)
     {
         /* cancel the running thread  */
         if (priv->health_check_active == _gf_true) {
-            pthread_cancel(priv->health_check);
+            ret = pthread_cancel(priv->health_check);
+            if (ret != 0) {
+                gf_msg(xl->name, GF_LOG_ERROR, 0, P_MSG_PTHREAD_CANCEL_FAILED,
+                       "Failed to send cancellation to health-check thread");
+                goto unlock;
```

Comment:
`ret` must be set to `-1` to avoid returning a positive value in case of error.

Reply:
Done